src/main/java/com/notably/model/block/BlockTreeItem.java

```diff
@@ -0,0 +1,61 @@
+package com.notably.model.block;
+
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import javafx.scene.control.TreeItem;
+
+/**
+ * API of the BlockNode component.
+ */
+public interface BlockTreeItem {
+    Title getTitle();
+
+    Body getBody();
```

Comment:
Here's why I say `BlockTreeItem` should extend `Block`!

1. `BlockTreeItem` supports the `Title` and `Body` method, both of which get delegated to the `Block` contained inside `BlockTreeItem`.
2. Conceptually-speaking, we can say that `BlockTreeItem` is a subtype of `Block`. That is, `BlockTreeItem` is just a specialised `Block` that supports observable tree operations. If we make `BlockTreeItem` extend `Block`, we'll allow the user of this API to do whatever they want to do with a `BlockTreeItem` as if it's a `Block`, which is great!

This pattern is actually called the [decorator pattern](https://en.wikipedia.org/wiki/Decorator_pattern)!